050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,744 @@
+
+The observer approach could work, but comes with some serious drawbacks:
+
+* The Kafka project does not provide a publicly-supported API for running a KRaft observer, so we'd need to make use of internal APIs which could change between releases without warning. 
+  This represents a potential maintenance burden to the project.
+* The operator would require persistent storage (for the replicated `__cluster_metadata` log), which would make the operator significantly more difficult to operate.
+  For example, if it were realized as a single pod `Deployment`, then the requirement for a persistent volume would tie it to a single AZ, meaning it would not be as highly available as the current operator.
+  This could be worked around by having multiple observers with only a single elected leader actually in charge of making modifications at any one time.
+
+Overall, it is clear that the complexity of the operator would be significantly increased by pursuing this direction.
+
+It's also worth reviewing _why_ people want to use an operator for topics. 
+There are two broad use cases:
+
+1. Declarative deployments, for example in the form of gitops. 
+2. Using `KafkaTopics` operationally as the API through which to manage all topics.
+
+Based on opened issues and Slack conversations with users, it seems that 1. is the most common reason people use the BTO.
+However, the BTO doesn't do a great job at satisfying this use case.
+
+* It only supports a single namespace, which means BTO admins cannot easily follow a workspace-per-dev team model.
+  It also means that infrastructure-level topics (e.g. `__consumer_offsets` and `__transaction_state`) comingle with per-application topics. 
+* It doesn't *enforce* a single source of truth, allowing for changes done directly in Kafka to conflict with declarative deployments. 
+
+Moreover, the BTO doesn't do a good job with item 2. either:
+
+* It only surfaces a subset of topic state through the Kube API (it doesn't include partition leadership/ISR for example)
+This is intentional (it doesn't make sense to overload etcd to present this kind of dynamic information), but it undermines the suitability of the BTO for this use case.
+
+For these reasons we are proposing the UTO.
+
+## Proposal
+
+### Goals
+
+* A unidirectional topic operator which is KRaft compatible
+
+### Non-goals
+
+* Supporting replication factor change (this could be added later)
+
+### Maintaining compatibility of the `KafkaTopic` custom resource
+
+We would remain schema-compatible with the existing `KafkaTopic` resource at the API level. 
+This means that all the existing `spec` sections could be handled by the new operator and all existing fields of the `status` section would still be present for applications that consume the status.
+However, the semantics of a `KafkaTopic` would change incompatibly: The `spec` would no longer be updated when Kafka-side configuration changed.
```

Comment:
When you say that the semantics of the `KafkaTopic` changes, I would imagine that the fields remain the same but have a different meaning. I do not think that is the case here -> the semantics of the KafkaTopic custom resource is still the same, and all the fields still have the same meaning. The backward incompatible change is from my point of view in the TO itself, not in the CR. So I think this could be a bit confusing.